src/App.tsx

```diff
@@ -1,14 +1,64 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { FilterTodos } from './types/FilterTodos';
+import { getTodos } from './api';
+
+const getFilteredTodos = (
+  todos: Todo[],
+  query: string,
+  selectedFilter: string,
+) => {
+  let todosCopy = [...todos];
+
+  if (query) {
+    const normalizedQuery = query.toLowerCase().trim();
+
+    todosCopy = todosCopy.filter(({ title }) => (
+      title.toLowerCase().includes(normalizedQuery)
+    ));
+  }
+
+  if (selectedFilter) {
+    todosCopy = todosCopy.filter(({ completed }) => {
+      switch (selectedFilter) {
+        case FilterTodos.Active:
+          return !completed;
+        case FilterTodos.Completed:
+          return completed;
+        default:
+          return todosCopy;
+      }
+    });
+  }
+
+  return todosCopy;
+};
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
```

Comment:
Better to set `false` as a default value and change it to `true` only before you make API call